doc/explanation/bootstrap.rst

```diff
@@ -0,0 +1,45 @@
+How Dune Uses Dune to Build Dune
```

Comment:
I love this header!

Reply:
that's a reference to https://zachholman.com/talk/how-github-uses-github-to-build-github/

Reply:
though we miss the triple meanint in the original ("how github (the company) uses github (the piece of software) to build github (the service)")